tests/aws/services/sqs/test_sqs_move_task.py

```diff
@@ -0,0 +1,452 @@
+            collected.extend(messages)
+
+            if delete:
+                for m in messages:
+                    sqs_client.delete_message(QueueUrl=queue_url, ReceiptHandle=m["ReceiptHandle"])
+
+        return len(collected) >= expected
+
+    if not poll_condition(_receive, timeout=timeout):
+        raise TimeoutError(
+            f"gave up waiting for messages (expected={expected}, actual={len(collected)}"
+        )
+
+    return collected
+
+
+def get_approx_number_of_messages(
+    sqs_client,
+    queue_url: str,
+) -> int:
+    response = sqs_client.get_queue_attributes(
+        QueueUrl=queue_url, AttributeNames=["ApproximateNumberOfMessages"]
+    )
+    return int(response["Attributes"]["ApproximateNumberOfMessages"])
+
+
+def sqs_wait_queue_size(
+    sqs_client,
+    queue_url: str,
+    expected_num_messages: int,
+    timeout: float = None,
+) -> int:
+    def _check_num_messages():
+        return get_approx_number_of_messages(sqs_client, queue_url) >= expected_num_messages
+
+    if not poll_condition(_check_num_messages, timeout=timeout):
+        raise TimeoutError(
+            f"gave up waiting for messages (expected={expected_num_messages}, "
+            f"actual={get_approx_number_of_messages(sqs_client, queue_url)})"
+        )
+
+    return get_approx_number_of_messages(sqs_client, queue_url)
+
+
+@pytest.fixture(autouse=True)
+def sqs_snapshot_transformer(snapshot):
+    snapshot.add_transformer(snapshot.transform.sqs_api())
+
+
+@pytest.fixture()
+def sqs_create_dlq_pipe(sqs_create_queue):
+    def _factory(max_receive_count: int = 1) -> tuple[QueueUrl, QueueUrl]:
+        dl_queue_url = sqs_create_queue()
+
+        # create redrive policy
+        url_parts = dl_queue_url.split("/")
+        dl_target_arn = arns.sqs_queue_arn(
+            url_parts[-1],
+            account_id=url_parts[len(url_parts) - 2],
+            region_name=TEST_AWS_REGION_NAME,
+        )
+        queue_url = sqs_create_queue(
+            Attributes={
+                "RedrivePolicy": json.dumps(
+                    {
+                        "deadLetterTargetArn": dl_target_arn,
+                        "maxReceiveCount": max_receive_count,
+                    }
+                )
+            },
+        )
+        return queue_url, dl_queue_url
+
+    return _factory
+
+
+@markers.aws.validated
+def test_cancel_with_invalid_task_handle(aws_client, snapshot):
+    with pytest.raises(ClientError) as e:
+        aws_client.sqs.cancel_message_move_task(TaskHandle="foobared")
+    snapshot.match("error", e.value.response)
+
+
+@markers.aws.validated
+def test_cancel_with_invalid_source_arn_in_task_handle(aws_client, snapshot):
+    source_arn = "arn:aws:sqs:us-east-1:878966065785:test-queue-doesnt-exist-123456"
+    task_handle = encode_move_task_handle("10f57157-fc38-4da9-a113-4de7e12d05dd", source_arn)
+
+    with pytest.raises(ClientError) as e:
+        aws_client.sqs.cancel_message_move_task(TaskHandle=task_handle)
```

Comment:
nit: we are relying on the fact here that the queue gets checked before the task. To be 100% precise, we would have to create a task that generates a valid handle, then test with a non-existent queue and said handle. But I think we can accept this here.

Reply:
arguably, this is even better since it tests parity of the order of AWS validations :smile: